### ISSUE TITLE
Backup failed: Text file busy

### ISSUE THREAD

**Issue by drjonnicholson** [spatie/laravel-backup#373]
I've just installed 4.4.9 into a Laravel 5.3 app, currently running in Homestead 4.0.4 on a Windows 10 host.

The backup is created, and successfully writes the zip to the filesystem, but fails when deleting the temporary file.

Here is the output to the console:
...$ php artisan backup:run
Starting backup...
Dumping database XXX...
Determining files to backup...
Zipping XXX files...
Created zip containing XXX files. Size is XXX MB
Copying zip to disk named XXX...
Successfully copied zip to disk named XXX.
Backup failed because: unlink(XXX/storage/app/laravel-backup/temp/2017-02-07-18-26-08.zip): Text file busy.

Any thoughts?

**Comment by drjonnicholson** [comment]
Tried on our live server, and it works fine. Just in Homestead I consistently have this problem.

**Comment by AlexVanderbist** [comment]
I cannot seem to replicate this issue on Laravel 5.4 or 5.3 in my Homestead (Windows 10 host). Are you sure it's not the host's file system keeping the temporary file busy? Maybe something like an editor trying to index the file or an antivirus program?

**Comment by EventFellows** [comment]
Same problem here as what @drjonnicholson desribes. Also win 10 with L5.3 and Backup 4.4.9

```
Starting backup...
Dumping database xxxxxx...
Determining files to backup...
Zipping 242 files...
Created zip containing 242 files. Size is 214.39 KB
Copying zip to disk named s3_backups...
Successfully copied zip to disk named s3_backups.
Backup failed because: unlink(/home/vagrant/Code/xxxxxx/storage/app/laravel-backup/temp/2017-02-09-16-46-01.zip): Text file busy.
```

Backup on done and saved to AWS S3 correctly.

Just selected the `app` folder for backup to test the backup. so don't worry about small file size.

**Comment by freekmurze** [comment]
Could you try to run `lsof` to see which process is accessing that file?

See: http://stackoverflow.com/questions/16764946/what-generates-the-text-file-busy-message-in-unix

**Comment by drjonnicholson** [comment]
This is the result of `lsof` from inside Vagrant around the time of the error:
```
vagrant@XXX:~/XXX$ lsof ~/XXX/ | grep laravel-backup
php     1904 vagrant    6u   REG   0,42 333518668 32993 /home/vagrant/XXX/storage/app/laravel-backup/temp/2017-02-11-15-39-41.zip
```
After the error happens `lsof` doesn't list anything.

From the windows host I've used the sysinternals Process Explorer tool to check what files are open, and nothing about laravel is listed. I also disabled anti-virus on the host machine.

**Comment by drjonnicholson** [comment]
Maybe a possible solution, but in your temporary directory package could code like that in Laravel's Filesystem be used (https://github.com/laravel/framework/blob/4.2/src/Illuminate/Filesystem/Filesystem.php#L108) to do the unlink without causing the exception, and then doing something like sleep for a short time before trying again? Then after x number of attempts it could then fail - so that execution isn't blocked if someone did open the file locally. Thoughts?

**Comment by freekmurze** [comment]
I really don't want to call `sleep` in the code. That feels like a band aid solution to me.

**Comment by drjonnicholson** [comment]
Very true, blame it on early starts and busses.

So after a bit more digging... Looks like  copyToBackupDestinations Calls write in BackupDestination, which fopen's the file but doesnt fclose it? Is the fclose done anywhere else I'm not spotting?

**Comment by freekmurze** [comment]
The thing is that we can't reproduce this problem. Most people seems unaffected. If you find a good fix for this problem, please send a PR.

**Comment by drjonnicholson** [comment]
Some may not experience the problem, but from this thread it's clear some have.

I've just added `fclose($handle);` to `BackupDestination.php` to line 84 and it solved the problem. I've done a pull request.

**Comment by freekmurze** [comment]
Fix has been tagged as `v4.6.6`
